### ISSUE TITLE
Class serialization code generator

### ISSUE THREAD

**Issue by vogel76** [steemit/steem#1932]
To achieve most efficient serialization, dedicated code shall be generated to dump and load object contents, basing on class definitions.

**Comment by mvandeberg** [comment]
This has little impact on reindex times themselves. If we are going to spend time optimizing performance, reindex and resync times should be priority. We already have working serialization code. It may not be the most efficient, but it works and are there are more impactful changes that can be implemented with the dev time it will take to do this.

**Comment by theoreticalbts** [comment]
I'm working on support code which will eventually allow us to generate class serialization code.

**Comment by vogel76** [comment]
@mvandeberg @theoreticalbts serialize tool we would like to use has been already done years ago by us in our other project. Now we have just needed adopt it to your needs (ie remove dependencies to our internal modules, add cmakelists as we used other build system for unix). I think it is done in 90% atm. Tool after all such changes is still working correctly for simulator code as we have checked a 2 days ago. @mkochanowicz is working on incorporating it into your build system and testing according to your code.
I understand that this is part which is not needed immediately (as you have said there is code doing serialization). But it makes no sense to start same work by @theoreticalbts

**Comment by mvandeberg** [comment]
This is not a priority task right now. Please work on assigned issues.

**Comment by dnotestein** [comment]
@mvandeberg @theoreticalbts I agree with @vogel76 that it makes no sense to create a new set of code for this when we have working code that requires very few modifications to use right now. I also don't agree that this isn't a priority task: it will allow for easy removal of chainbase, which is a huge resource hog right now. The main thing is that this can be done relatively quickly. This is the primary issue that I would like to see addressed before further work on SMTs. And it won't take very long to do at all.
